fastchat/model/model_adapter.py

```diff
@@ -504,10 +507,43 @@ def match(self, model_path: str):
 
     def get_default_conv_template(self, model_path: str) -> Conversation:
         return get_conv_template("h2ogpt")
+    
+class LoraAdapter(BaseAdapter):
+    def match(self, model_path: str):
+        # check if has adapter config in model_path
+        res = os.path.exists(os.path.join(model_path, "adapter_config.json"))
+        if res and 'lora' not in model_path.lower():
+            print("Warning: there is an adapter_config.josn at model_path, but model_path.lower() does not contain 'lora' in its name. This may not be a LoRA model.")
+        return res
+
+    def load_model(self, lora_path: str, from_pretrained_kwargs: dict):
+        from peft import PeftModel
+        lora_conf = json.load(open(os.path.join(lora_path, "adapter_config.json")))
+
+        base_model_path = lora_conf['base_model_name_or_path']
+
+        print(f"Loading the base model from {base_model_path}")
+        # base = AutoModelForCausalLM.from_pretrained(
+        #     base_model_path, torch_dtype=torch.float16, low_cpu_mem_usage=True
+        # )
+        # base_tokenizer = AutoTokenizer.from_pretrained(base_model_path, use_fast=False)
+
+        base, tokenizer = super().load_model(base_model_path, from_pretrained_kwargs)
```

Comment:
Do not call `super()` directly. Please call `def get_model_adapter(model_path: str) -> BaseAdapter` to get the correct adapter. I think you also need to prevent infinite recursion by excluding `LoraAdapter`

Reply:
+1 to that strategy! I'd really like to see this kind of high level wrapper added in.  I've so far manually hacked it in for myself but it's not as flexible as this.